homeassistant/components/persistent_notification/trigger.py

```diff
@@ -0,0 +1,79 @@
+"""Offer persistent_notifications triggered automation rules."""
+from __future__ import annotations
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_PLATFORM
+from homeassistant.core import CALLBACK_TYPE, HassJob, HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.trigger import TriggerActionType, TriggerData, TriggerInfo
+from homeassistant.helpers.typing import ConfigType
+
+from . import Notification, UpdateType, async_register_callback
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ("persistent_notification",)
+
+CONF_NOTIFICATION_ID = "notification_id"
+CONF_UPDATE_TYPE = "update_type"
+
+TRIGGER_SCHEMA = cv.TRIGGER_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_PLATFORM): "persistent_notification",
+        vol.Optional(CONF_NOTIFICATION_ID): str,
+        vol.Optional(CONF_UPDATE_TYPE): vol.All(
+            cv.ensure_list, [vol.Coerce(UpdateType)]
+        ),
+    }
+)
+
+
+async def async_attach_trigger(
+    hass: HomeAssistant,
+    config: ConfigType,
+    action: TriggerActionType,
+    trigger_info: TriggerInfo,
+) -> CALLBACK_TYPE:
+    """Listen for state changes based on configuration."""
+    trigger_data: TriggerData = trigger_info["trigger_data"]
+    job = HassJob(action)
+
+    persistent_notification_id = config.get(CONF_NOTIFICATION_ID)
+    persistent_notification_update_types = config.get(CONF_UPDATE_TYPE)
+
+    @callback
+    def persistent_notification_listener(
+        update_type: UpdateType, notifications: dict[str, Notification]
+    ) -> None:
+        """Listen for persistent_notification updates."""
+
+        for notification in list(notifications.values()):
```

Comment:
```suggestion
        for notification in notifications.values():
```

I don't think you need to make a copy unless there is a risk of the data changing during iteration